From d06d79e534d30b02632e2e4ed2bf5a061c96b9fe Mon Sep 17 00:00:00 2001
From: Marcus Williams <marcus.g.williams@gmail.com>
Date: Thu, 28 Dec 2017 09:25:40 -0800
Subject: [PATCH] Add fedora port to install

Signed-off-by: Marcus Williams <marcus.g.williams@gmail.com>
---
 install-ncsdk.sh | 36 ++++++++++++++++++++++++++++++++++--
 1 file changed, 34 insertions(+), 2 deletions(-)

diff --git a/install-ncsdk.sh b/install-ncsdk.sh
index 8b84aec..640b0b1 100755
--- a/install-ncsdk.sh
+++ b/install-ncsdk.sh
@@ -50,6 +50,8 @@ elif [ "${OS_DISTRO,,}" == "raspbian" ] && [ $OS_VERSION -ge 80 ] && [ $OS_VERSI
 		)
 	printf "$ERR_MSG"
 	exit 1
+elif [ "${OS_DISTRO,,}" == "fedora" ] && [ $OS_VERSION -ge 27 ]; then
+	echo "Installing on Fedora 27"
 else
 	echo "Your current combination of linux distribution and distribution version is not officially supported!"
 	exit 1
@@ -233,6 +235,15 @@ function exit_on_error
 	fi
 }
 
+function install_fedora_dependencies
+{
+	# Install dnf dependencies
+	exit_on_error "$SUDO_PREFIX dnf update -y"
+	exit_on_error "$SUDO_PREFIX dnf install -qy unzip coreutils curl git python3 python3-pip"
+	exit_on_error "$SUDO_PREFIX dnf install -qy $(cat "$DIR/requirements_dnf.txt")"
+	exit_on_error "$SUDO_PREFIX dnf install -qy boost-devel"
+}
+
 # Ubuntu dependencies handler
 function install_ubuntu_dependencies
 {
@@ -280,7 +291,7 @@ function find_tensorflow ()
 	fi
 }
 
-function install_ubuntu_python_dependencies
+function install_python_dependencies
 {
         echo "Installing python dependencies..."
 
@@ -399,6 +410,10 @@ function check_and_install_caffe()
 		CAFFE_SRC="https://github.com/intel/caffe.git"
 		CAFFE_VER="1.0.3"
 		CAFFE_DIR=intel-caffe
+	elif [ $CAFFE_FLAVOR == "intel" ] && [ "${OS_DISTRO,,}" == "fedora" ]; then
+                CAFFE_SRC="https://github.com/intel/caffe.git"
+                CAFFE_VER="1.0.3"
+                CAFFE_DIR=intel-caffe
 	elif [ $CAFFE_FLAVOR == "ssd" ]; then
 		CAFFE_SRC="https://github.com/weiliu89/caffe.git"
 		CAFFE_VER=""
@@ -424,6 +439,20 @@ function check_and_install_caffe()
 					return 0
 				fi
 			fi
+		elif [ $CAFFE_FLAVOR == "intel" ] && [ "${OS_DISTRO,,}" == "fedora" ]; then
+                        find_and_try_adjusting_symlinks
+
+                        # Look for intel caffe specific operation to determine the version of caffe currently running
+                        if [[ $? -eq 0 && -d "caffe" ]]; then
+                                cd caffe
+                                ./build/tools/caffe time -model models/bvlc_googlenet/deploy.prototxt -engine "MKLDNN" -iterations 1 &> /dev/null
+                                if [ $? -eq 1 ]; then
+                                        echo "Intel caffe requested but not found, installing Intel caffe..."
+                                else
+                                        echo "Intel caffe already installed, skipping..."
+                                        return 0
+                                fi
+                        fi
 		else
 			find_and_try_adjusting_symlinks
 			if [ $? -eq 0 ]; then
@@ -520,7 +549,7 @@ if [ $INSTALL_TOOLKIT == 'yes' ]; then
 	############################## Install dependencies #######################################
 	if [ "${OS_DISTRO,,}" == "ubuntu" ] || [ $OS_VERSION == 1604 ]; then
 		install_ubuntu_dependencies
-		install_ubuntu_python_dependencies
+		install_python_dependencies
 	elif [ "${OS_DISTRO,,}" == "raspbian" ] && [ $INSTALL_TENSORFLOW == "yes" ]; then
 		ERR_MSG=$(cat <<- END
 			${YELLOW}NOTE: Tensorflow is not officially supported on
@@ -530,6 +559,9 @@ if [ $INSTALL_TOOLKIT == 'yes' ]; then
 		printf "$ERR_MSG"
 		install_raspbian_dependencies
 		install_raspbian_python_dependencies
+	elif [ "${OS_DISTRO,,}" == "fedora" ] && [ $OS_VERSION == "27" ]; then
+		install_fedora_dependencies
+		install_python_dependencies
 	fi
 
 	check_and_install_caffe
-- 
2.14.3

